### DISCUSSION TITLE
合并 emoji 图片

### DISCUSSION THREAD

**Discussion by Tim-Paik** [walinejs/waline#589] [Q&A] [answered]
现在的版本中，每打开一个评论页面都会有非常多的关于 emoji 的请求
如图
![image](https://user-images.githubusercontent.com/49786711/135857769-c36a73f5-1a8d-4c24-b878-6ff559dc965a.png)
是否应该将这些请求的图片都合并为一个png？前端通过切图的方式显示
由于服务器TTFB以及HTTP并发连接限制的原因，分成如此多请求可能会降低性能
~~不过这可能是个 breaking change~~

**Comment by Mister-Hope** [comment]
~~原则上这些emoji是lazy load的，在支持的浏览器上只会在你点击对应选项卡的时候才会开始加载。此外lazy load 的图片会在页面内部的link访问完成后才会开始加载，不会阻塞浏览器网络。~~

~~另外由于waline本身获取评论延迟就不低 (由于官方推荐或者说默认的leancloud加vercel属于服务端和数据库分离的情况，所以服务端需要等待leancloud的网络请求完毕后才会返回)，所以评论本身的图片也不太可能会阻塞网页请求。~~

所以这里我个人不觉得需要有什么改变。

如果你确定在近两年的浏览器上无法延迟加载，请留言或开一个新的讨论来汇报，我们会处理。

P.S.: 我不太确定评论的emoji是否需要懒加载，目前印象中并不是，可以讨论一下。

---

更新：似乎默认选项卡的确会直接加载，原因是emoji已经出现在dom中了，这个后续会优化一下。

**Comment by Mister-Hope** [comment, marked as answer]
应该在最新版本被修复了
